lightning/src/ln/channelmanager.rs

```diff
@@ -10502,6 +10502,26 @@ where
 			"Dual-funded channels not supported".to_owned(),
 			 msg.channel_id.clone())), *counterparty_node_id);
 	}
+
+	fn handle_message_received(&self) {
+		let invoice_requests = self.pending_outbound_payments.get_invoice_request_awaiting_invoice();
+		if invoice_requests.is_empty() {
+			return;
+		}
+		let mut pending_offers_messages = self.pending_offers_messages.lock().unwrap();
+		for (payment_id, invoice_request) in invoice_requests {
+			let context = OffersContext::OutboundPayment { payment_id };
+			let reply_path = match self.create_blinded_path(context) {
+				Ok(path) => path,
+				Err(_) => return,
+			};
```

Comment:
Should we call this outside the loop? Then we can just clone the reply path or iterate over the returned ones once #3087 is landed. That way we don't need to hold the lock while computing the blinded paths.

Reply:
Sounds like a great idea!

Update handle_message_received to calculate reply_paths prior to locking the mutex in [2128346](https://github.com/lightningdevkit/rust-lightning/commit/21283464f7a3c32499d57c6e24ce32687c4e6394)
Thanks!